infra/vsts_agent_linux_startup.sh

```diff
@@ -95,6 +96,94 @@ useradd \
 #add docker group to user
 usermod -aG docker vsts
 
+# The Bazel cache fills up the hard drive in less than a day, and is a real
+# pain to clean up, as it consists of a very large amount of read-only files in
+# a fairly large tree of read-only directories. Therefore, it looks like the
+# best way to manage it is to mount it on a partition and reset the partition.
+
+cat <<REMOUNT_CACHE > /root/remount_cache.c
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+int exists(const char* path) {
+	struct stat s;
+	return stat(path, &s) == 0 && S_ISDIR(s.st_mode);
+}
+
+int execute_f(char** args) {
+	int status;
+	char** child_env = { NULL };
+	pid_t pid = fork();
+	if (0 == pid) {
+		// child process
+		if (-1 == execve(args[0], args, child_env)) {
+			printf("Failed to start child process calling %s.", args[0]);
+			perror(0);
+			exit(1);
+		}
+		// unreachable: if execve succeeded, we're off to another executable
+	}
+	// parent
+	if (-1 == pid) {
+		printf("Failed to fork.");
+		perror(0);
+		exit(1);
+	}
+	if (-1 == waitpid(pid, &status, 0)) {
```

Comment:
do we have to worry about orphaning the child?

Reply:
Maybe. I don't expect this to run more than twice on the lifetime of a machine if everything goes well; if anything goew wrong, I don't expect we'll be able to use the machine (as its disk is already full), so one zombie process isn't going to make much of a difference.